src/prefect/flow_engine.py

```diff
@@ -490,18 +490,6 @@ def start(self):
             self._client = client_ctx.sync_client
             self._is_started = True
 
-            # this conditional is engaged whenever a run is triggered via deployment
-            if self.flow_run_id and not self.flow:
-                self.flow_run = self.client.read_flow_run(self.flow_run_id)
-                try:
-                    self.flow = self.load_flow(self.client)
-                except Exception as exc:
-                    self.handle_exception(
-                        exc,
-                        msg="Failed to load flow from entrypoint.",
-                    )
-                    self.short_circuit = True
-
```

Comment:
What was all this about?  Is it not possible to get here without a `flow` anymore?

Reply:
This code is redundant because we fetch the flow run outside of the engine and pass it into the engine here: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/engine.py#L37-L42

Reply:
Amazing!

Reply:
Also, `flow` is a required attribute for the `FlowRunEngine` class.